sls-versions/src/main/java/com/palantir/sls/versions/VersionComparator.java

```diff
@@ -16,95 +16,53 @@
 
-import static com.palantir.logsafe.Preconditions.checkArgument;
-
-import com.palantir.logsafe.SafeArg;
 import java.util.Comparator;
-import java.util.OptionalInt;
 
 /** Compares {@link OrderableSlsVersion}s by "newness", i.e., "1.4.0" is greater/newer/later than "1.2.1", etc.. */
 public enum VersionComparator implements Comparator<OrderableSlsVersion> {
     INSTANCE;
 
     @Override
-    public int compare(OrderableSlsVersion left, OrderableSlsVersion right) {
-        if (left.getValue().equals(right.getValue())) {
+    @SuppressWarnings("ImmutablesReferenceEquality")
+    public int compare(OrderableSlsVersion version1, OrderableSlsVersion version2) {
+        // Short-circuit if the same instance is passed in
+        if (version1 == version2) {
             return 0;
         }
 
-        int mainVersionComparison = compareMainVersion(left, right);
-        if (mainVersionComparison != 0) {
-            return mainVersionComparison;
-        }
-
-        if ((left.getType() == SlsVersionType.RELEASE) || (right.getType() == SlsVersionType.RELEASE)) {
-            // Releases always compare correctly just by type now we know base version matches.
-            return Integer.compare(left.getType().getPriority(), right.getType().getPriority());
-        }
-
-        if ((left.getType() == SlsVersionType.RELEASE_SNAPSHOT)
-                && (right.getType() == SlsVersionType.RELEASE_SNAPSHOT)) {
-            // If both are snapshots, compare snapshot number.
-            return compareFirstSequenceVersions(left, right);
+        int result = Integer.compare(version1.getMajorVersionNumber(), version2.getMajorVersionNumber());
+        if (result != 0) {
+            return result;
         }
 
-        if (left.getType() == SlsVersionType.RELEASE_SNAPSHOT) {
-            // Snapshot is larger.
-            return 1;
+        result = Integer.compare(version1.getMinorVersionNumber(), version2.getMinorVersionNumber());
+        if (result != 0) {
+            return result;
         }
 
-        if (right.getType() == SlsVersionType.RELEASE_SNAPSHOT) {
-            // Snapshot is larger.
-            return -1;
-        }
-
-        return compareSuffix(left, right);
-    }
-
-    private int compareMainVersion(OrderableSlsVersion left, OrderableSlsVersion right) {
-        if (left.getMajorVersionNumber() != right.getMajorVersionNumber()) {
-            return left.getMajorVersionNumber() > right.getMajorVersionNumber() ? 1 : -1;
+        result = Integer.compare(version1.getPatchVersionNumber(), version2.getPatchVersionNumber());
+        if (result != 0) {
+            return result;
         }
 
-        if (left.getMinorVersionNumber() != right.getMinorVersionNumber()) {
-            return left.getMinorVersionNumber() > right.getMinorVersionNumber() ? 1 : -1;
+        // If RC version is not present, treat it as meaning positive infinite,
+        //   as all releases/release snapshots are considered newer than any RC.
```

Comment:
nit: Avoid abbreviations and ambiguous `/` usage

```suggestion
        // If release candidate version is not present, treat it as the maximum value.
        // All release and release snapshots versions are newer than any release candidate or release candidate snapshot version.
```